src/mint/Mint.ts

```diff
@@ -394,6 +395,56 @@ class Mint {
 		return response;
 	}
 
+	/**
+	 * Requests the mint to pay a melt offer.
+	 *
+	 * @remarks
+	 * This generic method allows any melt meltPayload that conforms to NUT-05.
+	 * @param meltPayload Payload containing quote ID, inputs, and custom REQ items.
+	 * @param options.customRequest Optional override for the request function.
+	 * @param options.preferAsync Optional override to set 'respond-async' header.
+	 * @returns Payment result with state and optional RES items.
+	 */
+	async melt<TRes extends Record<string, unknown> = Record<string, unknown>>(
+		method: string,
+		meltPayload: NUT05MeltPayload,
+		options?: {
+			customRequest?: RequestFn;
+			preferAsync?: boolean;
+		},
+	): Promise<NUT05MeltQuoteResponse & TRes> {
+		// Set headers as needed
+		const headers: Record<string, string> = {
+			...(options?.preferAsync ? { Prefer: 'respond-async' } : {}),
+		};
+		// Sanitize method string and make request
+		if (!/^[a-z0-9-]+$/.test(method)) {
```

Comment:
Reviewer note: This was changed in #414 to include underscore, and abstracted to a helper ready for work on generics